pb/context.go

```diff
@@ -0,0 +1,57 @@
+package pb
+
+import (
+	"context"
+)
+
+type ctxKey int
+
+// logFieldsKeyContext is the key that holds the log fields in a context.
+const logFieldsKeyContext ctxKey = 0
+
+// Fields is a log fields container
+type Fields map[string]interface{}
+
+func copyFields(fields Fields) Fields {
+	copy := make(Fields, len(fields))
+	return updateFields(copy, fields)
+}
+
+func updateFields(fields Fields, newFields Fields) Fields {
+	for k, v := range newFields {
+		if _, ok := fields[k]; ok {
+			continue
+		}
+
+		fields[k] = v
+	}
+
+	return fields
+}
+
+// GetLogFields returns a copy of the log fields of the context. It can be nil.
+func GetLogFields(ctx context.Context) Fields {
+	if v := ctx.Value(logFieldsKeyContext); v != nil {
+		return copyFields(v.(Fields))
+	}
+
+	return nil
+}
+
+// SetLogFields returns a context by updating the current log fields with those
+// provided. Setting a key that is already present has no effect.
```

Comment:
just wondering, is there a reason why you chose to ignore present keys instead of replacing it with the new values?

Reply:
ok I see now the reason is this discussion https://github.com/src-d/lookout-sdk/pull/77#discussion_r259796563, and it makes sense to preserve the fields set by lookout.

Reply:
shouldn't it be `AddLogFields` if we don't allow to change current fields?